mqtt-v5-broker/src/main.rs

```diff
@@ -181,24 +175,21 @@ fn init_logging() {
     }
 }
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
     init_logging();
 
-    // Creating a Runtime does the following:
-    // * Spawn a background thread running a Reactor instance.
-    // * Start a ThreadPool for executing futures.
-    // * Run an instance of Timer per thread pool worker thread.
-    let runtime = Runtime::new()?;
-
     let broker = Broker::new();
     let broker_tx = broker.sender();
-    runtime.spawn(broker.run());
+    let broker = task::spawn(async {
+        broker.run().await;
+        Ok(())
+    });
```

Comment:
I haven't used tokio in quite awhile, is this more idiomatic than `tokio::spawn(broker.run())`? Or was this just to get type-checking working for the call to `try_join_all`?